parsers/NTESMO.py

```diff
@@ -177,93 +204,154 @@ def parse_consumption(
 
 
 def parse_production_mix(
-    raw_production_mix: pd.DataFrame, logger: Logger
+    raw_production_mix_df: pd.DataFrame, logger: logging.Logger
 ) -> list[dict]:
-    production_mix = []
-    generation_units = set(raw_production_mix.columns)
-    generation_units.remove("Period Start")
-    generation_units.remove("Period End")
-    if not generation_units == PLANT_MAPPING.keys():
-        raise ParserException(
-            "NTESMO.py",
-            f"New generator {generation_units - PLANT_MAPPING.keys()} detected in AU-NT, please update the mapping of generators.",
-        )
-    raw_production_mix["Period Start"] = raw_production_mix[
+    raw_production_mix_df["Period Start"] = raw_production_mix_df[
         "Period Start"
     ].dt.tz_localize("Australia/Darwin")
-    for _, production in raw_production_mix.iterrows():
-        data_point = {
-            "zoneKey": "AU-NT",
-            "datetime": production["Period Start"].to_pydatetime(),
+
+    # some older data may miss columns for newer plants
+    raw_production_mix_df = raw_production_mix_df.dropna(axis=1, how="all")
+
+    generation_units = set(raw_production_mix_df.columns)
+    generation_units.remove("Period Start")
+    generation_units.remove("Period End")
+    unknown_units = generation_units.difference(PLANT_MAPPING.keys())
+    logger.warning(
+        f"New generator(s) {unknown_units} detected in AU-NT, please update the mapping of generators."
+    )
```

Comment:
This should probably be guarded behind an if statement to check if the set is empty or not first.